### PR TITLE
Fix sequence matching.

### LINKED ISSUE #660: Patmat: sequence matching is bugged
```
[info] # partest /Users/dark/workspace/dotty/tests/partest-generated/run/t889.scala
[info] % dotc t889.scala
[info] % java \
[info]   -classpath \
[info]   /Users/dark/workspace/dotty/tests/partest-generated/run/t889-run.obj:/Users/dark/workspace/dotty/target/scala-2.11/dotty_2.11-0.1-SNAPSHOT.jar:/Users/dark/.ivy2/cache/me.d-d/scala-compiler/jars/scala-compiler-2.11.5-20150506-175515-8fc7635b56.jar:/Users/dark/.ivy2/cache/org.scala-lang/scala-reflect/jars/scala-reflect-2.11.5.jar:/Users/dark/.ivy2/cache/org.scala-lang/scala-library/jars/scala-library-2.11.5.jar \
[info]   -Dfile.encoding=UTF-8 \
[info]   -Djava.library.path=/Users/dark/workspace/dotty/tests/partest-generated/run \
[info]   -Dpartest.output=/Users/dark/workspace/dotty/tests/partest-generated/run/t889-run.obj \
[info]   -Dpartest.lib=/Users/dark/.ivy2/cache/org.scala-lang/scala-library/jars/scala-library-2.11.5.jar \
[info]   -Dpartest.reflect=/Users/dark/.ivy2/cache/org.scala-lang/scala-reflect/jars/scala-reflect-2.11.5.jar \
[info]   -Dpartest.comp=/Users/dark/workspace/dotty/target/scala-2.11/dotty_2.11-0.1-SNAPSHOT.jar \
[info]   -Dpartest.cwd=/Users/dark/workspace/dotty/tests/partest-generated/run \
[info]   -Dpartest.test-path=/Users/dark/workspace/dotty/tests/partest-generated/run/t889.scala \
[info]   -Dpartest.testname=t889 \
[info]   -Djavacmd=java \
[info]   -Djavaccmd=javac \
[info]   -Duser.language=en \
[info]   -Duser.country=US \
[info]   scala.tools.nsc.MainGenericRunner \
[info]   -usejavacp \
[info]   Test \
[info]   jvm > t889-run.log
[info] scala.MatchError: List(a) (of class scala.collection.immutable.$colon$colon)
[info] 	at Test$.<init>(t889.scala:12)
[info] 	at Test$.<clinit>(t889.scala)
[info] 	at Test.main(t889.scala)
[info] 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[info] 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[info] 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[info] 	at java.lang.reflect.Method.invoke(Method.java:497)
[info] 	at scala.reflect.internal.util.ScalaClassLoader$$anonfun$run$1.apply(ScalaClassLoader.scala:70)
[info] 	at scala.reflect.internal.util.ScalaClassLoader$class.asContext(ScalaClassLoader.scala:31)
[info] 	at scala.reflect.internal.util.ScalaClassLoader$URLClassLoader.asContext(ScalaClassLoader.scala:101)
[info] 	at scala.reflect.internal.util.ScalaClassLoader$class.run(ScalaClassLoader.scala:70)
[info] 	at scala.reflect.internal.util.ScalaClassLoader$URLClassLoader.run(ScalaClassLoader.scala:101)
[info] 	at scala.tools.nsc.CommonRunner$class.run(ObjectRunner.scala:22)
[info] 	at scala.tools.nsc.ObjectRunner$.run(ObjectRunner.scala:39)
[info] 	at scala.tools.nsc.CommonRunner$class.runAndCatch(ObjectRunner.scala:29)
[info] 	at scala.tools.nsc.ObjectRunner$.runAndCatch(ObjectRunner.scala:39)
[info] 	at scala.tools.nsc.MainGenericRunner.runTarget$1(MainGenericRunner.scala:65)
[info] 	at scala.tools.nsc.MainGenericRunner.run$1(MainGenericRunner.scala:87)
[info] 	at scala.tools.nsc.MainGenericRunner.process(MainGenericRunner.scala:98)
[info] 	at scala.tools.nsc.MainGenericRunner$.main(MainGenericRunner.scala:103)
[info] 	at scala.tools.nsc.MainGenericRunner.main(MainGenericRunner.scala)
```

### PR DESCRIPTION
**Pull request by DarkDimius**
@odersky please review.
Fixes #660

### CODE REVIEW

**Comment by DarkDimius** [comment]
Unintentionally fixed #656.

**Comment by DarkDimius** [comment]
I believe we have some issue with partest here. Partest reports test failure:

But if I go to the directory with object files compiled by partest `/Users/dark/workspace/dotty/tests/partest-generated/run/t5629b-run.obj`

and execute the very same command that partest did:
```
java \
>    -classpath \
>    /Users/dark/workspace/dotty/tests/partest-generated/run/t5629b-run.obj:/Users/dark/workspace/dotty/target/scala-2.11/dotty_2.11-0.1-SNAPSHOT.jar:/Users/dark/.ivy2/cache/me.d-d/scala-compiler/jars/scala-compiler-2.11.5-20150506-175515-8fc7635b56.jar:/Users/dark/.ivy2/cache/org.scala-lang/scala-reflect/jars/scala-reflect-2.11.5.jar:/Users/dark/.ivy2/cache/org.scala-lang/scala-library/jars/scala-library-2.11.5.jar \
>    -Dfile.encoding=UTF-8 \
>    -Djava.library.path=/Users/dark/workspace/dotty/tests/partest-generated/run \
>    -Dpartest.output=/Users/dark/workspace/dotty/tests/partest-generated/run/t5629b-run.obj \
>    -Dpartest.lib=/Users/dark/.ivy2/cache/org.scala-lang/scala-library/jars/scala-library-2.11.5.jar \
>    -Dpartest.reflect=/Users/dark/.ivy2/cache/org.scala-lang/scala-reflect/jars/scala-reflect-2.11.5.jar \
>    -Dpartest.comp=/Users/dark/workspace/dotty/target/scala-2.11/dotty_2.11-0.1-SNAPSHOT.jar \
>    -Dpartest.cwd=/Users/dark/workspace/dotty/tests/partest-generated/run \
>    -Dpartest.test-path=/Users/dark/workspace/dotty/tests/partest-generated/run/t5629b.scala \
>    -Dpartest.testname=t5629b \
>    -Djavacmd=java \
>    -Djavaccmd=javac \
>    -Duser.language=en \
>    -Duser.country=US \
>    scala.tools.nsc.MainGenericRunner \
>    -usejavacp \
>    Test
```

The test runs correctly.

**Comment by odersky** [comment]
When I try manually it I get the same error as partest.

File: src/dotty/tools/dotc/transform/PatternMatcher.scala [hunk 1]
```diff
@@ -136,7 +136,7 @@ class PatternMatcher extends MiniPhaseTransform with DenotTransformer {thisTrans
         def callDirect   = tgt.select(nme.drop).appliedTo(Literal(Constant(n)))
         def callRuntime  = ref(defn.traversableDropMethod).appliedTo(tgt, Literal(Constant(n)))
 
-        def needsRuntime = tgt.tpe derivesFrom defn.SeqClass /*typeOfMemberNamedDrop(tgt.tpe) == NoType*/
+        def needsRuntime = !(tgt.tpe derivesFrom defn.SeqClass) /*typeOfMemberNamedDrop(tgt.tpe) == NoType*/
```
Comment (odersky) [inline src/dotty/tools/dotc/transform/PatternMatcher.scala]: Redundant parens around the expression.

**Comment by odersky** [comment]
Otherwise LGTM.

File: src/dotty/tools/dotc/transform/PatternMatcher.scala (hunk 1 above)
Comment (odersky) [inline src/dotty/tools/dotc/transform/PatternMatcher.scala]: Sorry, I was myopic here. All's good.

**Pull request merged**